lib/ansible/modules/cloud/amazon/aws_api_gateway_facts.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: aws_api_gateway_facts
+version_added: '2.9.0'
+short_description: Gather facts about API Gateways
+description: Gather facts about API Gateways
+author:
+  - Nao Morikawa (@cahlchang)
+requirements: [ boto3 ]
+
+extends_documentation_fragment:
+    - aws
+    - ec2
+options:
+  api_id:
+    description:
+      - The Amazon API Gateway ID.
+    required: false
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: gather facts about all API Gateway
+  aws_api_gateway_facts:
+    profile: YOUR_PROFILE
+
+- name: gather facts about an API Gateway using apigateway-id.
+  aws_api_gateway_facts:
+    api_id: xxxxxxxxxx
+'''
+
+RETURN = '''
+apis:
+  description: a list of api-gateway
+  returned: always
+  type: complex
+  contains:
```

Comment:
This does not seem to be at all comprehensive compared to what is in https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigateway.html#APIGateway.Client.get_rest_apis

You can generate the RETURN list using hacking/return_skeleton_generator.py